runners/direct-java/src/main/java/org/apache/beam/runners/direct/CommittedResult.java

```diff
@@ -49,12 +50,23 @@
    */
   public abstract Iterable<? extends CommittedBundle<?>> getOutputs();
 
+  /**
+   * Returns if the transform that produced this result produced outputs.
+   *
+   * <p>Transforms that produce output via modifying the state of the runner (e.g.
+   * {@link CreatePCollectionView}) should explicitly set this to true. If {@link #getOutputs()}
+   * returns a nonempty iterable, this will also return true.
+   */
+  public abstract boolean producedOutputs();
```

Comment:
Changed this to a set of types of output this transform produced.

`WriteView` adds `PCollection View` to the types of outputs. Bundles are controlled by the evaluation context.